compliance/azure/azure_low_cpu_usage/Azure CPU Below Utilization.pt

```diff
@@ -0,0 +1,222 @@
+    query "api-version","2016-09-01"
+    header "User-Agent", "RS Policies"
+  end 
+  result do
+    encoding "json"
+        field "vmname", val(iter_item,"name") 
+        field "rg", val(iter_item, "rg")
+        field "averages" do
+          collect jmes_path(response,"value[].data[]") do
+            field "average", jmes_path(col_item,"average")
+         end
+         end
+  end
+end
+
+datasource "ds_combineddata" do
+  run_script $js_combineddata,$param_cpuaveragepercentage,$ds_azure_instance_performance
+end
+
+#This is to get the average cpu over time for the machine and return the dataset of machines with issues
+script "js_combineddata", type: "javascript" do
+parameters "param_cpuaveragepercentage","ds_azure_instance_performance"
+result "result"
+code <<-EOS
+var result = []
+for (var i = 0; i < ds_azure_instance_performance.length; i++) {
+  
+  var machinename = ds_azure_instance_performance[i].vmname
+  var rg = ds_azure_instance_performance[i].rg
+  var averages = ds_azure_instance_performance[i].averages
+  var averagecpu 
+  for (var z = 0; z < averages.length; z++){
+    var averagenumber =averages[z]
+    averagecpu=0
+    flagtrip=0
+    if (!isNaN(averagenumber["average"]))
+      {
+    averagecpu = averagenumber["average"] + averagecpu
+    flagtrip=1
+  }
+  }
+
+ finalaverage = averagecpu / (averages.length + 1)
+ if((finalaverage < param_cpuaveragepercentage) && (flagtrip==1)){
+    server_json = { "vmname": machinename, "averagecpu": finalaverage, "rg": rg}
+    result.push(server_json)
+ }
+}
+EOS
+
+end
+
+
+policy "azure_resource_policy" do
+  validate $ds_combineddata do
+    summary_template "Azure Resources out of compliance"
+    detail_template <<-EOS
+# Azure Resources out of compliance
+| Name | CPU Average | Resource Group
+| ----------------------- | -----------------------  | -------------------- 
+{{ range data -}}
+{{.vmname}} | {{.averagecpu}} | {{.rg}}
+{{ end -}}
+EOS
+    check eq(size(data), 0)
+    escalate $email
+    escalate $delete_resources
+  end
+end
+
+escalation "email" do
+   email $param_email
+end
+
+
+define delete_resources($data, $param_azure_tenant_id,$param_azure_sub) return $all_responses do
+  call get_access_token($param_azure_tenant_id) retrieve $access_token
+  $all_responses = []
+  foreach $item in $data do
+   sub on_error: skip do
+      $response = http_request(
+        verb: "delete",
+        host: "management.azure.com",
+        href: join(["/subscriptions/", $param_azure_sub, "/resourceGroups/", $item["rg"], "/providers/Microsoft.Compute/virtualMachines/",$item["vmname"]]),
+        https: true,
+        query_strings: {
+          "api-version": "2018-06-01"
+        },
+        headers: {
+          "cache-control": "no-cache",
+          "content-type": "application/json",
+          "authorization": "Bearer " + $access_token
+        }
+      )
```

Comment:
log the response, and treat error to log definition.

Reply:
Done